File: factorial using recursion/factorialUsingRecursion.js
//parameterised way
// const factorial=(f,n)=>{
//     if(n===0 ||n===1){
//         return f;
//     }
//     return factorial(f*n,n-1);
// }
// let n=5;
// let f=1;
// console.log(factorial(f,n));

//functional way
const factorial =(n)=>{
    if(n===0 || n===1){
        return 1;
    }
    return n*factorial(n-1)
}
let n=5;
console.log(factorial(n));

File: factorial using recursion/factorialusingRecursion.py
# paramterised way
# def factorial(fac,i):
#     if i==0 or i==1:
#         return fac
#     return factorial(fac*i,i-1)

# n=5
# fac=1
# print(factorial(fac,n))

#functional way
def factorial(n):
    if(n==0 or n==1):
        return 1
    return n*factorial(n-1)

n=5
print(factorial(n))